python-package/lightgbm/basic.py

```diff
@@ -779,6 +788,7 @@ def create_valid(self, data, label=None, weight=None, group=None,
                       weight=weight, group=group, silent=silent, params=params,
                       free_raw_data=self.free_raw_data)
         ret._set_predictor(self._predictor)
```

Comment:
@guolinke why uses _set_predictor here and ._predictor = ... in subset()? The result should be the same?